libraries/loggly_helpers.rb

```diff
@@ -0,0 +1,41 @@
+module LogglyHelpers
+  def configure_files(original_files)
+    valid_files(original_files).map(&:to_h).map do |file|
+      file['tag'] ||= File.basename(file['filename']).tr('.', '-')
+      file['statefile'] ||= "#{file['filename']}.rsyslog_state"
+      file
+    end
+  end
+
+  private def valid_files(original_files)
+    original_files.reject do |file|
+      !file.is_a?(Hash) || file['filename'].to_s.strip.empty?
+    end
+  end
+
+  def loggly_format(app_name)
+    app_name_initial_caps = app_name.to_s.gsub(/\b(?<!['â`])[a-z]/, &:capitalize)
+    "LogglyFormat#{app_name_initial_caps}"
+  end
+
+  def tags
+    (node.loggly.tags || []).map { |tag| "tag=\\\"#{tag}\\\"" }.join(' ')
+  end
+
+  def rsyslog_major_version
+    version_cmd = Mixlib::ShellOut.new('rpm -q rsyslog')
+    version_cmd.run_command
+    response = version_cmd.stdout.chomp
+    # will return something like rsyslog-5.8.10-10.el6_6.x86_64
+    response.match(/^rsyslog-(\d+)/)[1].to_i
+  end
+
+  def set_version6_source(template, source_file)
+    resource = run_context.resource_collection.find(template: template)
+    resource.source(source_file)
```

Comment:
It could be more clear, yes.  I think it's preferable to allow configuration of this setting as a node attribute.  We can set a default value in attributes.rb based on what we know about what current platform versions install by default.  Centos 6.x installs 5.8.  Centos 7 installs 7.something.  Not sure about ubuntu 14.04.

The user could then override it their own wrapper cookbooks if they're installing a non-standard version.

I'll take a stab at this approach, and make sure that a user-provided value doesn't get clobbered by the attribute defaults.